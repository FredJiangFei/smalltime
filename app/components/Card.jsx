import React from 'react';
import {
  StyleSheet,
  TouchableWithoutFeedback,
  View,
  Image,
  Text,
} from 'react-native';
import colors from '../config/colors';

export default function Card({ desc, imageUrl, onPress }) {
  return (
    <TouchableWithoutFeedback onPress={() => onPress(desc)}>
      <View style={styles.root}>
        <Text>{desc}</Text>
        <Image style={styles.image} source={{ uri: imageUrl }} />
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  root: {
    backgroundColor: colors.secondary,
    marginTop: 8
  },
  image: {
    width: '100%',
    height: 50,
  },
});
