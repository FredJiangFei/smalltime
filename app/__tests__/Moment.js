import { render } from '@testing-library/react-native';
import Card from '../components/Card';

test('render card', () => {
  // Arrange
  const { getByText } = render(<Card desc="123" />);

  // Assert
  expect(getByText('123'));
});

test('render card snapshot', () => {
  // Arrange
  const card = render(
    <Card desc="123" imageUrl="https://picsum.photos/200/300" />
  );

  // Assert
  expect(card).toMatchSnapshot();
});
