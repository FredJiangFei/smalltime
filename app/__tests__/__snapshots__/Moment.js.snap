// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render card snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "red",
      "borderBottomWidth": 1,
      "marginTop": 8,
      "padding": 8,
      "paddingBottom": 8,
    }
  }
>
  <Text>
    123
  </Text>
  <Image
    source={
      Object {
        "uri": "https://picsum.photos/200/300",
      }
    }
    style={
      Object {
        "height": 150,
        "resizeMode": "contain",
        "width": "100%",
      }
    }
  />
</View>
`;
